plasma/root_chain/contracts/RootChain/RootChain.sol

```diff
@@ -200,42 +177,44 @@ contract RootChain {
     function challengeExit(uint256 cUtxoPos, uint256 eUtxoPos, bytes txBytes, bytes proof, bytes sigs, bytes confirmationSig)
         public
     {
+        var txList = txBytes.toRLPItem().toList(11);
+        // Checks that spent input submit is the same as the one being exited
+        require(txList[0].toUint() + txList[1].toUint() + txList[2].toUint() == eUtxoPos);
         uint256 txindex = (cUtxoPos % 1000000000) / 10000;
         bytes32 root = childChain[cUtxoPos / 1000000000].root;
-        uint256 priority = exitIds[eUtxoPos];
         var txHash = keccak256(txBytes);
         var confirmationHash = keccak256(txHash, root);
         var merkleHash = keccak256(txHash, sigs);
-        address owner = exits[priority].owner;
+        address owner = exits[eUtxoPos].owner;
 
         require(owner == ECRecovery.recover(confirmationHash, confirmationSig));
         require(merkleHash.checkMembership(txindex, root, proof));
-        delete exits[priority];
-        delete exitIds[eUtxoPos];
+        delete exits[eUtxoPos].owner;
```

Comment:
Why only delete the owner field instead of the whole object?

Reply:
We only delete the owner because we still need to track that the exit happened so that it can't be resubmitted.

Reply:
In this case we should be making sure that the owner is not `0x0` in `finalizeExits`, otherwise we're just burning money, even when an exit has been successfully challenged.